corehq/apps/change_feed/pillow.py

```diff
@@ -9,16 +9,18 @@
 from corehq.apps.change_feed.topics import get_topic
 from couchforms.models import all_known_formlike_doc_types
 import logging
+from pillowtop.checkpoints.manager import PillowCheckpoint, get_django_checkpoint_store
 from pillowtop.couchdb import CachedCouchDB
 from pillowtop.listener import PythonPillow
 
 
 class ChangeFeedPillow(PythonPillow):
 
-    def __init__(self, couch_db, kafka):
+    def __init__(self, couch_db, kafka, checkpoint):
         super(ChangeFeedPillow, self).__init__(couch_db=couch_db)
         self._kafka = kafka
         self._producer = KeyedProducer(self._kafka)
+        self._checkpoint = checkpoint
```

Comment:
should be cleaned up a bit in https://github.com/dimagi/commcare-hq/pull/9046 and https://github.com/dimagi/pillowtop/pull/176

also i'm brain dumping some high-level tasks/todos here in case you want to review or follow along https://docs.google.com/document/d/1c9e12reAd0VSgAnWcZPMzKZzhQqU4N1I8UwUANv5iaA/edit